web/app/scripts/src/directives/calculate-percentage.es.js

```diff
@@ -29,10 +29,12 @@ module.directive('calculatePercentage', [
                 });
 
                 // Using true to compare the sub-elements.
-                scope.$watch('model', (answers) => {
-                    const numberOfCompletedQuestions = CalculateCompletion.countNumberOfAnsweredQuestions(answers);
-                    scope.percentageComplete = CalculateCompletion.calculatePercentageComplete(numberOfCompletedQuestions, totalQuestionCount);
-                }, true);
+                if (totalQuestionCount > 0) {
```

Comment:
The `link` is only run once (per directive on the page) so I would expect `totalQuestionCount` to always be `0` during the `link` function. I think you need to put the optimisation in the watch:
```
                scope.$watch('model', (answers) => {
                    if (totalQuestionCount == 0) {
                      return;
                    }
                    const numberOfCompletedQuestions = CalculateCompletion.countNumberOfAnsweredQuestions(answers);
                    scope.percentageComplete = CalculateCompletion.calculatePercentageComplete(numberOfCompletedQuestions, totalQuestionCount);
                }, true);
```